### DISCUSSION TITLE
How to approach bug identification in Fossify Calendar?

### DISCUSSION THREAD

**Discussion by jaskonius** [FossifyOrg/General-Discussion#470] [Q&A]
Hi!

I've recently started using Calendar paired with DAVx5 and Radicale. All worked quite nicely until, a few days ago, all Monday events of a specific calendar vanished. All other events (seem) not to be affected, just all events of this particular calendar which occur on a Monday. 

When I try to refresh, it mostly just loops infinitely. Sometimes, however, I get an "java....IllegalArgumentException" printed in a toast. The argument does not fit in the message toast but it appears to begin with a timestamp ("20240612...").

Until now, I've
- rechecked in Thunderbird (no problems there)
- reinstalled Calendar through fdroid
- installed Etar through fdroid, no problems there

So it _seems_ to be a bug in Calendar, which I'd like to hunt down to create a proper issue.

So, tl;dr: How can I accumulate more information on this problem to provide more information in an issue? I'd be fine using proper development tools but I've never really done Android related stuff so any hints would be welcome!

**Comment by naveensingh** [comment]
Please create a bug report including as much information as possible: https://github.com/FossifyOrg/Calendar/issues/new/choose

It would really help if you can share the affected events as an ICS file in your bug report. Apart from that, you don't really have to do anything. We'll ask you if we need any more information to reproduce the problem.

**Comment by viharm** [comment]
Is this behaviour intentional?

If the target has the following datasets as a result of multiple backups for multiple sources:
1. BackupNode/Tank/Server1Backup
2. BackupNode/Tank/Server2Backup
3. BackupNode/Tank/Laptop3Backup

... then, when backing up to Server2Backup, is it necessary for the thinner to traverse the other trees to find snapshots to thin? Is it not possible to restrict this to the base target dataset specified in the command (BackupNode/Tank) plus a combination of the source dataset & `--strip-path`?
